backend/file-ucloud-service/src/jvmMain/kotlin/services/FileScanner.kt

```diff
@@ -0,0 +1,505 @@
+                            username
+                        }
+                        else {
+                            log.warn("Skipping $username")
+                            null
+                        }
+                    }
+                } else chunk
+
+                while (true) {
+                    collectionsList.zip(filteredChunk.sortedBy { it }).forEach { (coll, path) ->
+                        withContext(pool) {
+                            launch {
+                                val file =
+                                    pathConverter.relativeToInternal(RelativeInternalFile("/home/$path"))
+                                submitScan(file, coll)
+                            }.join()
+                        }
+                    }
+
+                    if (resolvedCollections.next == null) {
+                        return@forEach
+                    }
+                    resolvedCollections = retrieveCollections(
+                        providerGenerated = true,
+                        includeOthers = true,
+                        chunk.map { PathConverter.COLLECTION_HOME_PREFIX + it },
+                        resolvedCollections.next
+                    )
+                }
+            }
+
+            println("Scanning Project folders")
+            project.chunked(50).forEach { chunk ->
+                var resolvedCollections = retrieveCollections(
+                    providerGenerated = true,
+                    includeOthers = true,
+                    chunk.map { PathConverter.COLLECTION_PROJECT_PREFIX + it },
+                    next = null
+                )
+                val collectionsList = resolvedCollections.items.sortedBy { it.owner.createdBy }
+                val filteredChunk = if (collectionsList.size != chunk.size) {
+                    log.warn("Might be missing collection. Number of collections: ${collectionsList.size}. Chunk size: ${chunk.size}")
+
+                    val projectIds = collectionsList.map{ it.owner.project }
+                    chunk.mapNotNull { projectFolder ->
+                        if (projectIds.contains(projectFolder)) {
+                            projectFolder
+                        } else {
+                            log.warn("Skipping $projectFolder")
+                            null
+                        }
+                    }
+
+                } else chunk
+
+                while (true) {
+                    collectionsList.zip(filteredChunk).forEach { (coll, path) ->
+                        withContext(pool) {
+                            launch {
+                                val file =
+                                    pathConverter.relativeToInternal(RelativeInternalFile("/projects/$path}"))
+                                submitScan(file, coll)
+                            }.join()
+                        }
+                    }
+
+                    if (resolvedCollections.next == null) {
```

Comment:
Won't this always be the case if we just do `itemsPerPage = 50`?

Reply:
I also expected this to alway be null, but this is the place where sometimes it runs one more turn.
I can delete it if you want

Reply:
We can keep it, but I suspect it is not needed.